Search/Metadata/ClassMetadata.php

```diff
@@ -78,18 +78,41 @@ public function getIndexMetadata($contextName)
         return $this->indexMetadatas[$contextName];
     }
 
-    public function serialize()
+    public function __serialize(): array
+    {
+        $data = [
+            $this->name,
+            $this->methodMetadata,
+            $this->propertyMetadata,
+            $this->fileResources,
+            $this->createdAt,
+        ];
+
+        return [$data, $this->indexMetadatas, $this->repositoryMethod];
+    }
+
+    public function __unserialize(array $data): void
     {
-        $data = parent::serialize();
+        list($data, $indexMetadata, $this->repositoryMethod) = $data;
+        $this->indexMetadatas = $indexMetadata;
 
-        return \serialize([$data, \serialize($this->indexMetadatas), $this->repositoryMethod]);
+        [
+            $this->name,
+            $this->methodMetadata,
+            $this->propertyMetadata,
+            $this->fileResources,
+            $this->createdAt,
+        ] = $data;
```

Comment:
same as above calling `parent::__unserialize` didn't work like expected so we did reimplment the parent method call.